lib/services/ngsi/ngsiService.js

```diff
@@ -184,11 +184,23 @@ function addTimestamp(payload) {
         };
 
     function addMetadata(attribute) {
+        var timestampFound = false;
+
         if (!attribute.metadatas) {
             attribute.metadatas = [];
         }
 
-        attribute.metadatas.push(timestamp);
+        for (var i = 0; i < attribute.metadatas.length; i++) {
+            if (attribute.metadatas[i].type === constants.TIMESTAMP_TYPE &&
+                attribute.metadatas[i].name === constants.TIMESTAMP_ATTRIBUTE) {
+                attribute.metadatas[i].value = timestamp.value;
+                timestampFound = true;
```

Comment:
You could `break` the loop to avoid further processing once the `timestamp` metadata entry is found and its value updated :)